src/pipecat/services/together.py

```diff
@@ -206,9 +205,24 @@ async def process_frame(self, frame: Frame, direction: FrameDirection):
             context = frame.context
         elif isinstance(frame, LLMMessagesFrame):
             context = TogetherLLMContext.from_messages(frame.messages)
-        elif isinstance(frame, LLMModelUpdateFrame):
-            logger.debug(f"Switching LLM model to: [{frame.model}]")
-            self.set_model_name(frame.model)
+        elif isinstance(frame, LLMUpdateSettingsFrame):
+            if frame.model is not None:
+                logger.debug(f"Switching LLM model to: [{frame.model}]")
+                self.set_model_name(frame.model)
+            if frame.frequency_penalty is not None:
+                await self.set_frequency_penalty(frame.frequency_penalty)
+            if frame.max_tokens is not None:
+                await self.set_max_tokens(frame.max_tokens)
+            if frame.presence_penalty is not None:
+                await self.set_presence_penalty(frame.presence_penalty)
+            if frame.temperature is not None:
+                await self.set_temperature(frame.temperature)
+            if frame.top_k is not None:
+                await self.set_top_k(frame.top_k)
+            if frame.top_p is not None:
+                await self.set_top_p(frame.top_p)
+            if frame.extra:
+                await self.set_extra(frame.extra)
```

Comment:
Do you mind create a function for this so we only have:

```
        elif isinstance(frame, LLMUpdateSettingsFrame):
            await self._update_setting(frame)
```

Just trying to keep `process_frame()` as short as possible, because sometimes it can get too long.

Reply:
`await self._update_settings(frame)`  (plural)